primefaces-integration-tests/src/test/java/org/primefaces/integrationtests/fileupload/AbstractFileUploadTest.java

```diff
@@ -29,12 +29,23 @@
 import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Collectors;
+import javax.faces.context.FacesContext;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Assumptions;
 import org.primefaces.selenium.AbstractPrimePageTest;
 import org.primefaces.selenium.component.DataTable;
 import org.primefaces.selenium.component.model.datatable.Row;
 
 public abstract class AbstractFileUploadTest extends AbstractPrimePageTest {
+    /**
+     * The file upload tests only work in conjunction with Mojarra.
+     * Unfortunately MyFaces does not work.
+     * Call {@code assumeMojarra()} at the beginning of each file upload test.
+     * This method should be removed if file upload tests also work with MyFaces.
+     */
+    protected void assumeMojarra() {
+        Assumptions.assumeTrue(!FacesContext.class.getPackage().getImplementationVendor().contains("Apache"));
```

Comment:
@christophs78 there might be a better way to do this like your TestTag that only runs if Safari etc?  That way you can mark the whole class.

@tandraschko we should get to the bottom of why these tests don't work with MyFaces.

Reply:
I found [p:FileUPload doesn't work Primefaces + Quarkus + MyFaces](https://forum.primefaces.org/viewtopic.php?t=62002) discussion. So I added context parameter primefaces.UPLOADER = commons to web.xml (and `commons-fileupload` dependency). But no luck, it didn't work. Afterwards I decided to skip these tests in conjunction with MyFaces.